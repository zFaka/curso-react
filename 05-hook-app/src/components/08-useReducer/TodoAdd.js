import React, {useState} from 'react';
import './styles.css' ;
import useForm from '../../hooks/useForm';

const TodoAdd = ({handleAddTodo}) => {



  const [show, setShow] = useState(false);

  
  const [{description}, handleInputChange, reset] = useForm({

    description:''
  });



  const handleSubmit = (e) => {

    //Esto evita que recargue la pagina una vez hecho el submit

    e.preventDefault();

    if(description.trim().length <= 1){return}

    const newTodo = {

      id:new Date().getTime(), 
      desc:description, 
      done: false
    };

    handleAddTodo(newTodo);

    reset();
  }



  return (
    <>
      <form onSubmit={handleSubmit}>
        <input 
          type='text'
          name= 'description'
          className='form-control rounded'
          autoComplete='off'
          value={description}
          onChange={handleInputChange}
        />
        {
         show 
          && 
            <button 
              className='btn btn-primary'
              type='submit'
            >
              + 
            </button>
        }
      </form>
    </>
  )}

export default TodoAdd;
