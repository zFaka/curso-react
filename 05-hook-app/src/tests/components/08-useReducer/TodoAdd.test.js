import {shallow} from "enzyme";
import TodoAdd from "../../../components/08-useReducer/TodoAdd";

describe('Test TodoApp', () => {


  const handleAddTodo = jest.fn();

  const wrapper = shallow(
    <TodoAdd
      handleAddTodo={handleAddTodo}
    />); 




  test('Should render', () => {
    expect(wrapper)
      .toMatchSnapshot()
  })




  test('Should dont call handleAddTodo', () => {


    const formSubmit = wrapper.find('form').prop('onSubmit');

    formSubmit({preventDefault(){}})


    expect(handleAddTodo).toHaveBeenCalledTimes(0)
  })




  test('Should call handleAddTodo', () => {


    const value = 'lrn mny';


    // inputChange simulation     value:value = value
    wrapper.find('input').simulate('change', {
      target:{
        value, 
        name:'description'
      }}); 


    // submit simulation
    wrapper.find('form').simulate(
      'submit', 
      {preventDefault(){}});



    // call handleAddTodo
    expect(handleAddTodo)
      .toHaveBeenCalledTimes(1);

    // Its true if this is an object, 
    // can be empty and is wrong
    expect(handleAddTodo)
      .toHaveBeenCalledWith(expect.any(Object));

    // if the id is a number is true
    expect(handleAddTodo)
      .toHaveBeenCalledWith({

      id:expect.any(Number), 
      desc:value, 
      done:false
    })

    // check if reset() works and 
    // clear the input and its value
    expect(wrapper.find('input').prop('value'))
      .toBe('');
  })
})
