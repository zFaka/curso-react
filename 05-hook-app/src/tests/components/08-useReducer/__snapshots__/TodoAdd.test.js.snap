// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodoApp Should render 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control rounded"
      name="description"
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
